### ISSUE TITLE
APT repository signing key needs to be manually updated on client side in order to extend expiration date

### ISSUE THREAD

**Issue by zertrin** [keybase/client#9626]
A few hour ago I had:

```
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://dist.keybase.io/linux/deb/repo stable InRelease: The following signatures were invalid: EXPKEYSIG 47484E50656D16C7 Keybase.io Code Signing (v1) <code@keybase.io>
W: Failed to fetch http://dist.keybase.io/linux/deb/repo/dists/stable/InRelease  The following signatures were invalid: EXPKEYSIG 47484E50656D16C7 Keybase.io Code Signing (v1) <code@keybase.io>
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

It seems the initial key was created with an expiration date of 4 years, expiring on 2017-11-19 and that the expiration date has been extended for 14 more years very recently (on 2017-11-13): see output of `gpg --list-packets code_signing_key.asc` here: https://haste.zertrin.org/raw/golotegofa

The key added by apt-key add prior to 2017-11-13 will not have the new signature from current published GPG key, requiring user to understand what's going on (it took me almost an hour to figure it out, including the time to figure out how to print details from gpg commend about the key) and take manual action:

- download `code_signing_key.asc`
- `sudo apt-key add code_signing_key.asc`

Suggestion: better communicate when manual action is required, for example on https://keybase.io/blog

**Comment by maxtaco** [comment]
Cc: @oconnor663 can you provide some details here?

**Comment by oconnor663** [comment]
@zertrin yes this is my fault. We wrote some guidance but I didn't get it online in time. Fixing now,

**Comment by oconnor663** [comment]
Ok we have a doc up at https://keybase.io/docs/the_app/linux_expired_key.

**Comment by dominicsayers** [comment]
The document linked above is now a 404

**Comment by zertrin** [comment]
> 
> 
> The document linked above is now a 404

There's an archived version: https://web.archive.org/web/20200105201931/https://keybase.io/docs/the_app/linux_expired_key

**Comment by rpatterson** [comment]
And years later, now `# apt-key ...` is known to be insecure and deprecated. Took a bit of searching, but I got the key re-integrated into my local `/etc/apt/keyrings/keybase.gpg` following [@askeli's excellent StackExchange answer](https://askubuntu.com/a/1307181/16083). Seriously, one of the best StackExchange answers I've ever seen, elegantly educated me about things I didn't yet understand while solving my problem practically at the same time, *and* helped me improve my security for other APT sources while I was at it!

But yeah, Keybase's Linux install documentation is badly broken and desperately needs updating.
